Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1663,12 +1664,15 @@ SubstituteData FrameLoader::defaultSubstituteDataForURL(const URL& url)
 {
     if (!shouldTreatURLAsSrcdocDocument(url))
         return SubstituteData();
-    auto& srcdoc = m_frame->ownerElement()->attributeWithoutSynchronization(srcdocAttr);
+    RefPtr iframeElement = dynamicDowncast<HTMLIFrameElement>(m_frame->ownerElement());
+    ASSERT(iframeElement);
+
+    auto& srcdoc = iframeElement->attributeWithoutSynchronization(srcdocAttr);
```

Comment:
Seeing pointers assumed to be non-null with debug-only assertions makes me sad.

I know it's preexisting, but - since we're expanding the use of these possibly-null objects - let's take the opportunity to also clean this up a bit.

Let's either find a way to early return or decide that RELEASE_ASSERT is safe to do.

(FrameLoader.cpp is in 3 SaferCPPExpectations files, and I'm guessing some of this wouldn't fly if it were being checked.)

Reply:
In this case, quick way to achieve this is replacing `ASSERT` with early return.
```c++
    RefPtr iframeElement = dynamicDowncast<HTMLIFrameElement>(m_frame->ownerElement());
    if (!iframeElement)
        return SubstituteData();
```